x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportJobTaskAction.java

```diff
@@ -54,8 +52,6 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
         String jobId = request.getJobId();
         // We need to check whether there is at least an assigned task here, otherwise we cannot redirect to the
         // node running the job task.
-        ClusterState state = clusterService.state();
-        JobManager.getJobOrThrowIfUnknown(jobId, state);
```

Comment:
There is no way to check for the existence of the job config here. If an async call is made the function will exit and continue it's work and a blocking call should not be made here.

The change will mean that all actions that inherit `TransportJobTaskAction` will return an error along the lines of `the persistent task does not exist for job [unknown-job]` rather than the usual unknown job message.